corehq/apps/locations/bulk_management.py

```diff
@@ -189,15 +201,23 @@ def lookup_old_collection_data(self, old_collection):
 
         for attr in self.meta_data_attrs:
             setattr(self.db_object, attr, getattr(self, attr, None))
-        self.db_object.metadata = self.custom_location_data
+        if self.custom_data != self.NOT_PROVIDED or self.uncategorized_data != self.NOT_PROVIDED:
+            self.db_object.metadata = self.custom_location_data
 
     @property
     @memoized
     def custom_location_data(self):
-        custom_data = {}
-        custom_data.update(self.custom_data)
-        custom_data.update(self.uncategorized_data)
-        return custom_data
+        if self.custom_data is self.NOT_PROVIDED or self.uncategorized_data is self.NOT_PROVIDED:
+            # if either of these are not provided, then existing data should be updated, not overridden
+            metadata = copy.copy(self.db_object.metadata)
+        else:
+            # if both of these are provided, then existing data should be overridden
+            metadata = {}
+        if self.custom_data != self.NOT_PROVIDED:
+            metadata.update(self.custom_data)
+        if self.uncategorized_data != self.NOT_PROVIDED:
+            metadata.update(self.uncategorized_data)
+        return metadata
```

Comment:
Going to fully respond tomorrow, but I am doing the validation already [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/locations/bulk_management.py#L598), this method assumes validation is over